File: Modules/SEDO/EMS/SIE.Web.EMS/Faults/Commands/AddEquipLargeFaultCommand.js
SIE.defineCommand('SIE.Web.EMS.Faults.Commands.AddEquipLargeFaultCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit" },

    execute: function (view, source) {
        var entity = this.getEditEntity();
        view.execute({
            data: {},
            success: function (res) {
                entity.setCode(res.Result);
            }
        });
        this.edit(entity);
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS/Faults/Commands/AddEquipLargeFaultCommand.cs
using SIE.EMS.Faults;
using SIE.Web.Command;

namespace SIE.Web.EMS.Faults.Commands
{
    /// <summary>
    /// 添加故障大类
    /// </summary>
    [JsCommand("SIE.Web.EMS.Faults.Commands.AddEquipLargeFaultCommand")]
    public class AddEquipLargeFaultCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 获取设备大类编码
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return RT.Service.Resolve<EquipFaultController>().GetEquipLargeFaultCode();
        }
    }
}
